core/unix/loader.c

```diff
@@ -754,11 +755,13 @@ privload_search_rpath(privmod_t *mod, bool runpath, const char *name,
                              len - strlen(RPATH_ORIGIN) - pre_len,
                              origin + strlen(RPATH_ORIGIN));
                     NULL_TERMINATE_BUFFER(path);
-                    snprintf(filename, MAXIMUM_PATH, "%s/%s", path, name);
+                    if (!lib_found)
+                        snprintf(filename, MAXIMUM_PATH, "%s/%s", path, name);
                 } else {
                     snprintf(path, BUFFER_SIZE_ELEMENTS(path), "%.*s", len, list);
                     NULL_TERMINATE_BUFFER(path);
-                    snprintf(filename, MAXIMUM_PATH, "%s/%s", path, name);
+                    if (!lib_found)
+                        snprintf(filename, MAXIMUM_PATH, "%s/%s", path, name);
```

Comment:
This is getting complex and I find it confusing with line 766 below.  Since 758-759 and these two lines above are identical, let's move these out and down below and so combine 763,764,766,792,793 all inside if (!lib_found).  That should be much cleaner and readable and less "spaghetti".

Reply:
Done.